frontend/src/streamDetails/__tests__/StreamStatisticsCard.test.jsx

```diff
@@ -1,131 +1,165 @@
   let wrapper;
-  const dummyStream = firstStreamResponse;
+  const dummyStream = getSampleStream();
   const dummyStats = getSampleStreamStats();
-
   const expectedProperties = {
     Time: dummyStats.time,
     "Round-Trip-Time": dummyStats.link.rtt,
     "Packets Retransmitted": dummyStats.send.packetsRetransmitted,
     "Packets Dropped": dummyStats.send.packetsDropped
   };
-
   const expectedState = {
     stats: dummyStats
   };
-
   const expectedButton = new ButtonInfo(
     `/Streams/Details/${dummyStream.id}/Statistics`,
     { statistics: dummyStats, stream: dummyStream },
     "More Statistics"
   );
 
-  beforeEach(() => {
-    wrapper = Enzyme.shallow(<StreamStatisticsCard stream={dummyStream} />);
-  });
   afterEach(() => {
     wrapper.unmount();
+    jest.clearAllMocks();
   });
 
-  it("should have an empty default state", () => {
-    const startingState = {
-      stats: null
-    };
-    expect(wrapper.state()).toStrictEqual(startingState);
-  });
-
-  describe("handleStatsChange() function", () => {
-    it("should set the stats state", () => {
-      const randomPreviousState = {
-        stats: new StreamStatisticsInfo(
-          1000,
-          2000,
-          new StreamStatsWindowInfo(12, 13, 14),
-          new StreamStatsLinkInfo(15, 16, 17),
-          new StreamStatsSendInfo(21, 22, 23, 24, 25, 26, 27),
-          new StreamStatsReceiveInfo(31, 32, 33, 34, 35, 36, 37, 38, 39)
-        )
-      };
-      wrapper.instance().setState(randomPreviousState);
-      wrapper.instance().handleStatsChange(dummyStats);
-      expect(wrapper.state()).toStrictEqual(expectedState);
-    });
-  });
-  describe("getProperties() function", () => {
-    it("if state is null, should return an empty object", () => {
-      const emptyObject = {};
-      const properties = wrapper.instance().getProperties();
-      expect(properties).toStrictEqual(emptyObject);
+  describe("componentDidMount() function", () => {
+    beforeEach(() => {
+      wrapper = Enzyme.shallow(<StreamStatisticsCard stream={dummyStream} />, {
+        disableLifecycleMethods: true
+      });
     });
-    it("if state is not null, should return a specific array of properties", () => {
-      wrapper.instance().setState(expectedState);
-      const actualProperties = wrapper.instance().getProperties();
-      expect(actualProperties).toStrictEqual(expectedProperties);
+    describe("calls StreamApi.getStreamStatistics()", () => {
+      it("if it resolves, it then calls handleStatsChange()", async () => {
+        jest
+          .spyOn(StreamApi, "getStreamStatistics")
+          .mockResolvedValue(dummyStats);
+
+        const handleStatsChangeSpy = jest.spyOn(
+          wrapper.instance(),
+          "handleStatsChange"
+        );
+
+        wrapper.instance().componentDidMount();
+
+        expect(StreamApi.getStreamStatistics).toHaveBeenCalledWith(
+          dummyStream.id
+        );
+
+        await new Promise(setImmediate);
```

Comment:
What does that do? create a promise then wait for it to be resolved instantly?

Reply:
waits for all promises to resolve, necessary for the code under test to finish before we assert. (discussed in person)